src/ExadelMentorship.BusinessLogic/CommandInvoker.cs

```diff
@@ -0,0 +1,30 @@
+﻿using ExadelMentorship.BusinessLogic.Exceptions;
+using ExadelMentorship.BusinessLogic.Features;
+using ExadelMentorship.BusinessLogic.Features.WeatherFeature;
+using ExadelMentorship.BusinessLogic.Interfaces;
+using ExadelMentorship.BusinessLogic.Models;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ExadelMentorship.BusinessLogic
+{
+    public class CommandInvoker
+    {
+        private readonly IServiceProvider _serviceProvider;
+        public CommandInvoker(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public Task Invoke<T>(T command) where T : ICommand
```

Comment:
I was quite surprised by the simplicity of this! But then I remembered that the parser will be providing `ICommand` objects, not directly its implementing classes. 

So, it is better to make the method signature as `Task Invoke(ICommand command)`

Reply:
done. Please review again